src/org/opensha/commons/calc/magScalingRelations/magScalingRelImpl/TMG2017_MagAreaRel.java

```diff
@@ -0,0 +1,304 @@
+ * 
+ * Addition to rake, we also use regime (="crustal" or "interface") as an additional parameter 
+ * to distinguish shallow crustal and subduction-interface. The default is "crustal". 
+ * However, the relations for normal faults are also applicable to inslab events.   
+ *  
+ * Notes:
+ * (1) Valid rake is in the range -180 to 180 degrees.  
+ * (2) the standard deviation for area as a function of mag is given for log(area) (base-10) not area.
+ * (2) The current implementation does not handle fixed seismogenic width (or saturated rupture width);
+ *     It will require implementing MagLenRel and MagWidRel as well.   
+ *     
+ * Also see: https://github.com/thingbaijam/sceqsrc
+ *    
+ * </p>
+ *
+ * @version 0.0
+ */
+
+public class TMG2017_MagAreaRel extends MagAreaRelationship {
+    /**
+	 * To Do: Write something here.
+	 */
+	private static final long serialVersionUID = 1L;
+	final static String C = "TMG2017_MagAreaRel";
+    public final static String NAME = "Thingbaijam et al.(2017)";
+    protected String regime = "crustal"; 
+    
+    
+    /**
+     * set regime
+     * @param regime string either "crustal" or "interface"
+     */
+    public void setRegime(String regime) {
+        FaultUtils.assertValidRegime(regime);
+        this.regime = regime;
+      }
+    
+    
+    /**
+     * Computes the median magnitude from rupture area for previously set rake and regime values. 
+     * @param area in km
+     * @return median magnitude MW
+     */
+    public double getMedianMag(double area){
+     
+      // FaultUtils.assertValidRake(rake) does not invalidate NaN - :(   
+      
+      if (Double.isNaN(rake))
+        // return NaN because this estimate is not available.   
+        return  Double.NaN; 
+      
+      if (regime.equalsIgnoreCase("crustal"))
+      		if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+      			// strike slip
+      			return 3.701 + 1.062*Math.log(area)*lnToLog;
+      		else if (rake > 0)
+      			// shallow crustal thrust/reverse faulting
+      			return  4.158 + 0.953* Math.log(area)*lnToLog;
+      		else
+      			// normal faulting 
+      			return 3.157 + 1.238*Math.log(area)*lnToLog;
+      else
+    	// subduction-interface
+      	return 3.469 + 1.054*Math.log(area)*lnToLog;
+    }
+    
+    
+    /**
+     * Gives the standard deviation for the magnitude as a function of area
+     * for previously-set rake and regime values
+     * @param area in km
+     * @return standard deviation
+     */
+    public double getMagStdDev(){
+      if (Double.isNaN(rake))
+        // Not available
+        return Double.NaN;
+      
+      if (regime.equalsIgnoreCase("crustal"))
+    	if (( rake <= 45 && rake >= -45 ) || rake >= 135 || rake <= -135)
+    	  // strike slip
+    	  return  0.184;
+    	else if (rake > 0)
+    	  // thrust/reverse
+    	  return  0.121;
+    	else
+    	  // normal
+    	  return  0.181;
+      else
```

Comment:
for example, indentation and curly braces would make it very easy to see which `if` this `else` belongs to.

Reply:
okay. Done!